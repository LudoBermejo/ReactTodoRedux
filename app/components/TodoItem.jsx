import React from 'react';
import moment from 'moment';

const TodoItem = (props) => {
  const { id, value, completed, createdAt, completedAt } = props;
  const todoClassName = completed ? 'todo-item todo-completed' : 'todo-item';
  function renderCreatedAt(createdAt, completedAt) {
    if (!completedAt) {
      return `Created ${moment.unix(createdAt).format('DD MM YYYY | HH:mm')}`;
    } else {
      return `Completed ${moment.unix(completedAt).format('DD MM YYYY | HH:mm')}`;
    }
  }

  function onChangeComplete(e) {
    e.preventDefault();
    props.onChangeComplete(id);
  }
  return (
    <div
      id={`list_item_${id}`}
      onClick={onChangeComplete}
      className={todoClassName}
    >
      <div>
        <input
          type="checkbox"
          checked={completed}
        />
      </div>
      <div>
        <p>{value}</p>
        <p className="todo__subtext">{renderCreatedAt(createdAt, completedAt)}</p>
      </div>

    </div>
  );
};

TodoItem.propTypes = {
  id: React.PropTypes.string.isRequired,
  value: React.PropTypes.string.isRequired,
  completed: React.PropTypes.bool.isRequired,
  createdAt: React.PropTypes.number.isRequired,
  completedAt: React.PropTypes.number,
  onChangeComplete: React.PropTypes.func.isRequired
};

export default TodoItem;
