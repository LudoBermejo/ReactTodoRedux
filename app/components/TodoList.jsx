import React from 'react';
import TodoItem from 'TodoItem';

const TodoList = (props) => {
  const renderList = (listOfTodos) => {
    if(listOfTodos.length) {
      return listOfTodos.map(item => <TodoItem key={item.id} {...item} onChangeComplete={props.onChangeItemComplete} />);
    } else {
      return (
        <p className="container__message">No items to show</p>
      )
    }
  }

  const { todoList } = props;
  return (
    <div>
        {renderList(todoList)}
    </div>
  );
};

TodoList.propTypes = {
  todoList: React.PropTypes.array.isRequired,
  onChangeItemComplete: React.PropTypes.func.isRequired
};

export default TodoList;
