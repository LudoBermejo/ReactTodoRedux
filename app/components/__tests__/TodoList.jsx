import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import uuid from 'node-uuid';
import moment from 'moment';

import TodoList from 'TodoList';
import TodoItem from 'TodoItem';

describe('TodoList', () => {
  it('renders properly', () => {
    const values = [
      {
        id: uuid(),
        value: 'test_one',
        createdAt: moment().unix(),
        completed: false
      },
      {
        id: uuid(),
        value: 'test_two',
        createdAt: moment().unix(),
        completed: false
      }
    ];

    const todoList = renderer.create(<TodoList todoList={values} onChangeItemComplete={() => {}} />).toJSON();
    expect(todoList).toMatchSnapshot();
  });

  describe('List items properly', () => {
    it('renders all list of items', () => {
      const values = [
        {
          id: uuid(),
          value: 'test_one',
          createdAt: moment().unix(),
          completed: false
        },
        {
          id: uuid(),
          value: 'test_two',
          createdAt: moment().unix(),
          completed: false
        }
      ];

      const todoList = mount(<TodoList todoList={values} onChangeItemComplete={() => {}} />);
      expect(todoList.find(TodoItem).length).toBe(2);
    });

    it('shows message when no items in list', () => {
      const todoList = mount(<TodoList todoList={[]} onChangeItemComplete={() => {}} />);
      expect(todoList.find('p').length).toBe(1);
      expect(todoList.find(TodoItem).length).toBe(0);
    })
  });



});

