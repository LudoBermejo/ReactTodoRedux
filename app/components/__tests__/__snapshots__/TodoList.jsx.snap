exports[`TodoList renders properly 1`] = `
<div>
  <div
    className="todo-item"
    id="list_item_aae1ad3a-c65f-4535-a89a-52032a3ebdb5"
    onClick={[Function]}>
    <div>
      <input
        checked={false}
        type="checkbox" />
    </div>
    <div>
      <p>
        test_one
      </p>
      <p
        className="todo__subtext">
        Created 14 02 2017 | 13:14
      </p>
    </div>
  </div>
  <div
    className="todo-item"
    id="list_item_50a0a219-c64d-4842-81fb-210afa02df46"
    onClick={[Function]}>
    <div>
      <input
        checked={false}
        type="checkbox" />
    </div>
    <div>
      <p>
        test_two
      </p>
      <p
        className="todo__subtext">
        Created 14 02 2017 | 13:14
      </p>
    </div>
  </div>
</div>
`;
